numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -540,6 +540,15 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
         return ret;
     }
     else if (!PyBytes_Check(object) || PyBytes_GET_SIZE(object) < 1) {
+        int ret = DEPRECATE("Non-string object detected for " \
```

Comment:
Good point on the ```\```.  It's hard juggling multiple languages at once. :)  I did a slight revision to the warning messages to not explicitly mention the ```ORDER``` parameter